spec/jobs/back_in_stock_notification_email_job_spec.rb

```diff
@@ -26,19 +27,20 @@
         end
 
         context "for matching stock location name" do
-          let(:stock_location_params) { {name: bisn.stock_location.name} }
+          let(:stock_location_params) { {name: stock_location.name} }
 
-          it "sends the email" do
-            expect { subject }
-              .to change { Spree::BackInStockNotification.find(bisn.id).pending? }
-              .from(true).to(false)
+          it "calls the mailer job" do
+            subject
+            expect(BackInStockNotificationMailerJob).
+              to receive(:perform_later).
+              with(email: bisn.email, stock_location: stock_location)
```

Comment:
@jarednorman this currently fails. Am I missing something obvious?

Any idea how to run a specific test with Solidus extensions? I run them all with `bin/run` but can't find out how to limit tests to one line.

Reply:
To write the test this way you need to make the expectation before you call `subject`. The expectation has to be in place before the call so that it can be detected. Ruby is pretty magical, but not quite magical enough to look back in time and find out what methods got called with which arguments. ✨ 

My suggestion this morning (or afternoon for you) was to use [this matcher](https://relishapp.com/rspec/rspec-rails/docs/matchers/have-enqueued-job-matcher).

As for running individual specs, I think you should be able to do `bundle exec rspec spec/jobs/back_in_stock_notification_email_job_spec.rb:32` to run this particular spec. Alternatively, you could add `config.filter_run focus: true` to the RSpec config and then tag this test with `:focus`.

Reply:
Excellent.

I swear running rspec the usual way did not work before! Perhaps I fixed it when setting up CircleCI.